temporal-sdk/src/test/java/io/temporal/workflow/nexus/WorkflowOperationLinkingTest.java

```diff
@@ -0,0 +1,139 @@
+import io.temporal.client.WorkflowStub;
+import io.temporal.nexus.WorkflowClientOperationHandlers;
+import io.temporal.testing.internal.SDKTestWorkflowRule;
+import io.temporal.workflow.*;
+import io.temporal.workflow.shared.TestNexusServices;
+import io.temporal.workflow.shared.TestWorkflows;
+import java.time.Duration;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class WorkflowOperationLinkingTest extends BaseNexusTest {
+  @Rule
+  public SDKTestWorkflowRule testWorkflowRule =
+      SDKTestWorkflowRule.newBuilder()
+          .setWorkflowTypes(TestNexus.class, TestOperationWorkflow.class)
+          .setNexusServiceImplementation(new TestNexusServiceImpl())
+          .build();
+
+  @Override
+  protected SDKTestWorkflowRule getTestWorkflowRule() {
+    return testWorkflowRule;
+  }
+
+  @Test
+  public void testWorkflowOperationLinks() {
+    TestWorkflows.TestWorkflow1 workflowStub =
+        testWorkflowRule.newWorkflowStubTimeoutOptions(TestWorkflows.TestWorkflow1.class);
+    String result = workflowStub.execute(testWorkflowRule.getTaskQueue());
+    Assert.assertEquals("Hello from operation workflow " + testWorkflowRule.getTaskQueue(), result);
+    History history =
+        testWorkflowRule
+            .getWorkflowClient()
+            .fetchHistory(WorkflowStub.fromTyped(workflowStub).getExecution().getWorkflowId())
+            .getHistory();
+    // Assert that the operation started event has a link to the workflow execution started event
+    HistoryEvent nexusStartedEvent =
+        getEventOfType(history, EventType.EVENT_TYPE_NEXUS_OPERATION_STARTED);
+    Assert.assertEquals(1, nexusStartedEvent.getLinksCount());
+    Assert.assertEquals(
+        EventType.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED,
+        nexusStartedEvent.getLinks(0).getWorkflowEvent().getEventRef().getEventType());
+  }
+
+  public static class TestNexus implements TestWorkflows.TestWorkflow1 {
+    @Override
+    public String execute(String input) {
+      NexusOperationOptions options =
+          NexusOperationOptions.newBuilder()
+              .setScheduleToCloseTimeout(Duration.ofSeconds(10))
+              .build();
+      NexusServiceOptions serviceOptions =
+          NexusServiceOptions.newBuilder()
+              .setEndpoint(getEndpointName())
+              .setOperationOptions(options)
+              .build();
+      TestNexusServices.TestNexusService1 serviceStub =
+          Workflow.newNexusServiceStub(TestNexusServices.TestNexusService1.class, serviceOptions);
+      // Start an asynchronous operation backed by a workflow
+      NexusOperationHandle<String> asyncOpHandle =
+          Workflow.startNexusOperation(serviceStub::operation, input);
+      NexusOperationExecution asyncExec = asyncOpHandle.getExecution().get();
+      // Signal the operation to unblock, this makes sure the operation doesn't complete before the
+      // operation
+      // started event is written to history
```

Comment:
You can add a TODO to remove this once the server properly handles this race.

Reply:
Is there a github issue for this?

Reply:
Unfortunately no, we're doing most of this tracking with an internal tool. In any case we'll need to wait for the next server release.
@pdoerner I realized you'll need to fix this race in the test server too when you get to it but let's start with the real server.